src/include/replication/replica_replication_manager.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <string>
+
+#include "replication/replication_manager.h"
+#include "storage/recovery/replication_log_provider.h"
+#include "transaction/transaction_defs.h"
+
+namespace noisepage::replication {
+
+/**
+ * The replication manager that should run on the replicas.
+ *
+ * TODO(WAN): If the primary fails, how does the replica switch its replication manager? I guess possible, but annoying.
+ */
+class ReplicaReplicationManager final : public ReplicationManager {
+ public:
+  /**
+   * On construction, the replication manager establishes a listen destination on the messenger and connects to all the
```

Comment:
Does this mean that all the replicas connect to each other? If so, why is that?

Reply:
Yes. Currently, there is no reason to do so (replica1 never talks to replica2, for example). But you could imagine wanting to do fault tolerance, PAXOS, or something that requires talking to other replicas in the future.

Reply:
Marking resolved for now, feel free to reopen.